pypeclub/ayon-frontend
# Feature: Status Select Dropdown MVP

### Brief Description

Select and change an entity status by clicking on it and selecting from a dropdown.

### Description

- `StatusSelect` and `StatusField` components reusable across the app.
- New `OptionsDropdown` dropdown that's agnostic and reusable.
- `onChange` handlers post data and then reload data automatically.
- `StatusField` has `alignment` and `size` props.
- Uses pure CSS for all styling effects to optimise speed.
- Status reflects changes instantly (before db request has resolved).
- Status has a nice highlighted to fading off effect to emphasise a change.
- `StatusSelect` `size` changes based on status column size.
- Width is calculated using length of longest status.

![status_select_v003](https://user-images.githubusercontent.com/49156310/209951867-2075fcd4-1b87-4fd9-ad49-2a443ae93b9f.gif)

### TODO

- Icons default to circle when not found (API needs updating).
- If two of the same status are shown in different areas of the app, the none selected status won't auto update (RTK Query should fix).
- Multiselect (change multiple ones at once).

### Testing

- Click on any status and a dropdown should appear with all status options
- Clicking a new status should hide the dropdown and the original status should change.
- Try making the Status column bigger and smaller, `StatusSelect` should adapt to size.
